rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -4423,6 +4423,92 @@ public NewClass withArguments(JContainer<Expression> arguments) {
         }
     }
 
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    @With
+    class NullableType implements J, TypeTree, Expression {
+        @Nullable
+        @NonFinal
+        transient WeakReference<J.NullableType.Padding> padding;
+
+        @Getter
+        @EqualsAndHashCode.Include
+        UUID id;
+
+        @Getter
+        Space prefix;
+
+        @Getter
+        Markers markers;
+
+        @Getter
+        List<J.Annotation> annotations;
+
+        JRightPadded<TypeTree> typeTree;
+
+        public TypeTree getTypeTree() {
+            return typeTree.getElement();
+        }
+
+        @Override
+        public @Nullable JavaType getType() {
+            return typeTree.getElement().getType();
```

Comment:
Hmm. This will actually be wrong for Kotlin, as the nullability is lost in the type attribution. Possibly we already now have to add `JavaType.Nullable` or alternatively some kind of `nullable` flag.

Reply:
do you mean like add a `JavaType.NullableType` like this
```java
    @RequiredArgsConstructor
    @AllArgsConstructor(access = AccessLevel.PRIVATE)
    @Getter
    @With
    class NullableType implements JavaType {
        private JavaType raw;
    }
```
and changed here to

```java
        @Override
        public @Nullable JavaType getType() {
            return new JavaType.NullableType(typeTree.getElement().getType());
        }
```

Reply:
Either that (although for consistency it should probably just be called `Nullable`) or have it encoded in `JavaType.Class#flagsBitMap`. I think the latter might not work, as we can in Kotlin also have something like
```kt
fun <T> foo(): T? = ...
```
where a generic type variable is nullable.

But `Nullable` should also extend `JavaType.FullyQualified` otherwise there will be a lot of recipes that won't work anymore.

Let's see if we can get @sambsnyd to chime in on this.

Reply:
I think extending `JavaType$FullyQualified` (FQ) may lead to unexpected results when applied to a `JavaType$GenericTypeVariable` (GTV). The `FullyQualified` class contains a lot of methods that do not apply to a GTV. It's relatively common to check if the type is an `FQ` and progress further with a condition in a recipe, so I'd be cautious.

Reply:
Let's leave the type attribution as is for now. I am thinking we should maybe instead add annotations to the type attribution, as I suspect that is what Kotlin also uses in the bytecode and would also work for the various nullable annotations out there. But this is a bigger piece of work outside the scope of this PR.

Reply:
AFAICT the Java bytecode indeed contains `org.jetbrains.annotations.Nullable` annotations, so having that available in the type attribution is probably the way forward.